Factory/DiactorosFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Symfony\Bridge\PsrHttpMessage\Factory;
+
+use Symfony\Bridge\PsrHttpMessage\PsrHttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Zend\Diactoros\ServerRequest as DiactorosRequest;
+use Zend\Diactoros\Response as DiactorosResponse;
+use Zend\Diactoros\UploadedFile as DiactorosUploadedFile;
+
+/**
+ * Builds Psr\HttpMessage instances using the Zend Diactoros implementation.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class DiactorosFactory implements PsrHttpMessageFactoryInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createRequest(Request $symfonyRequest)
+    {
+        return new DiactorosRequest(
+
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createResponse(Response $symfonyResponse)
+    {
+        return new DiactorosResponse(
+
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createUploadedFile(UploadedFile $symfonyUploadedFile)
```

Comment:
Well, we could make it public later if we get requests about it. But for now, I would keep it private until we get more feedback (increasing the visibility of a method is easy. Decreasing it is forbidden outside major versions, and it is still a pain for major versions if we want to provide an easy upgrade)